[Bug]: rslint crashes after upgrading to v0.5

### System Info

  System:
    OS: Linux 6.17 Ubuntu 24.04.4 LTS 24.04.4 LTS (Noble Numbat)
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 6.57 GB / 15.38 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 24.11.0 - /run/user/1000/fnm_multishells/322307_1776921485301/bin/node
    npm: 11.6.1 - /run/user/1000/fnm_multishells/322307_1776921485301/bin/npm
    pnpm: 10.30.0 - /run/user/1000/fnm_multishells/322307_1776921485301/bin/pnpm
    bun: 1.3.10 - /home/nikhils/.bun/bin/bun
  Browsers:
    Chrome: 147.0.7727.101
  npmPackages:
    @rslint/core: ^0.5.0 => 0.5.0 

### Details

> rslint --type-check --quiet .

panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0x7e4faa]

goroutine 52793 [running]:
sync.(*WaitGroup).Go.func1.1()
        /opt/hostedtoolcache/go/1.26.0/x64/src/sync/waitgroup.go:251 +0x45
panic({0xcae400?, 0x1c25480?})
        /opt/hostedtoolcache/go/1.26.0/x64/src/runtime/panic.go:860 +0x13a
github.com/microsoft/typescript-go/internal/checker.(*Checker).resolveEntityName(0x0, 0x101f22889888, 0x1b3bf, 0x0?, 0x1, 0x101ef42ef4a0?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/checker/checker.go:15225 +0x28a
github.com/microsoft/typescript-go/internal/checker.(*Checker).getSymbolOfNameOrPropertyAccessExpression(0x0, 0x101f22889888)
        /home/runner/_work/rslint/rslint/typescript-go/internal/checker/checker.go:30794 +0x493
github.com/microsoft/typescript-go/internal/checker.(*Checker).getSymbolAtLocation(0x0, 0x101f22889888, 0x1)
        /home/runner/_work/rslint/rslint/typescript-go/internal/checker/checker.go:30598 +0xc67
github.com/microsoft/typescript-go/internal/checker.(*Checker).GetSymbolAtLocation(0x0, 0xd22400?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/checker/checker.go:30527 +0x2d
github.com/web-infra-dev/rslint/internal/utils.GetDeclaration(0x61eb58?, 0x101ef42ef618?)
        /home/runner/_work/rslint/rslint/internal/utils/ts_eslint.go:355 +0x13
github.com/web-infra-dev/rslint/internal/plugins/react/rules/no_danger_with_children.init.func1.1(0x101ef42ef618?)
        /home/runner/_work/rslint/rslint/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go:54 +0x2c
github.com/web-infra-dev/rslint/internal/plugins/react/rules/no_danger_with_children.init.func1.4(0x101ef42ef6c8?, {0xeb8036, 0x8})
        /home/runner/_work/rslint/rslint/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go:171 +0x189
github.com/web-infra-dev/rslint/internal/plugins/react/rules/no_danger_with_children.init.func1.6(0x101f22896b90, 0x101f22896b90, 0x101f59397b60?)
        /home/runner/_work/rslint/rslint/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go:207 +0xe8
github.com/web-infra-dev/rslint/internal/plugins/react/rules/no_danger_with_children.init.func1.8(0xcc1680?)
        /home/runner/_work/rslint/rslint/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go:231 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func8(0x11e, 0x101f22896b90)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:296 +0x5f
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f22896b90)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:343 +0x42
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*ParenthesizedExpression).ForEachChild(0x101f224772c0?, 0x101ef42ef7d8?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:7334 +0x23
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101f598e00da?, 0x101f22434a10?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f22434a10)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*ReturnStatement).ForEachChild(0x1?, 0x101efa15ad88?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:3301 +0x23
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101ef42e00fe?, 0x101f22793d88?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f22793d88)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*Block).ForEachChild(0x1?, 0x101f59545860)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:3764 +0x67
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101ef42e00f2?, 0x101f228985a0?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f228985a0)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*ArrowFunction).ForEachChild(0x101f224772c0, 0x101f59545860)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:6573 +0x152
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101f595400dc?, 0x101f224772c0?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f224772c0)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclaration).ForEachChild(0x101f2240f588, 0x101f59545860)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:3861 +0xc2
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xa50105?, 0x101f2240f588?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f2240f588)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclarationList).ForEachChild(0x0?, 0x101f59545860)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:3911 +0x67
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xec0106?, 0x101f22474f40?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f22474f40)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*VariableStatement).ForEachChild(0x101f228a0240?, 0x101f59545860?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:3807 +0x62
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101f599900f4?, 0x101f0386fbd0?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram.func10(0x101f0386fbd0)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:358 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*SourceFile).ForEachChild(0x101f2288bc08, 0x101f59545860)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:11191 +0x9b
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x101f2288bc08?, 0x0?)
        /home/runner/_work/rslint/rslint/typescript-go/internal/ast/ast.go:252 +0x1c
github.com/web-infra-dev/rslint/internal/linter.RunLinterInProgram(0x101ef8ca5448, {0x0, 0x0, 0x0}, {0x101eec4bee20, 0x1, 0x0?}, {0x1c29ea0, 0x2, 0x2}, ...)
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:366 +0x1152
github.com/web-infra-dev/rslint/internal/linter.RunLinter.func3()
        /home/runner/_work/rslint/rslint/internal/linter/linter.go:458 +0xb5
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        /home/runner/_work/rslint/rslint/typescript-go/internal/core/workgroup.go:40 +0x13
sync.(*WaitGroup).Go.func1()
        /opt/hostedtoolcache/go/1.26.0/x64/src/sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1
        /opt/hostedtoolcache/go/1.26.0/x64/src/sync/waitgroup.go:238 +0x73

### Reproduce link

_No response_

### Reproduce Steps

1. Install rslint@0.5.0
2. Run `rslint .`

--- a/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go
+++ b/internal/plugins/react/rules/no_danger_with_children/no_danger_with_children.go
@@ -51,6 +51,13 @@ var NoDangerWithChildrenRule = rule.Rule{
 			if ident == nil || ident.Kind != ast.KindIdentifier {
 				return nil
 			}
+			// TypeChecker is nil for gap files (files in the program but not
+			// in typeInfoFiles). Without it we cannot resolve an identifier to
+			// its declaration — skip spread/identifier lookups and fall back to
+			// structural checks only, rather than panic in GetDeclaration.
+			if ctx.TypeChecker == nil {
+				return nil
+			}
 			decl := utils.GetDeclaration(ctx.TypeChecker, ident)
 			if decl == nil || decl.Kind != ast.KindVariableDeclaration {
 				return nil

--- a/internal/plugins/react/rules/style_prop_object/style_prop_object.go
+++ b/internal/plugins/react/rules/style_prop_object/style_prop_object.go
@@ -39,6 +39,12 @@ var StylePropObjectRule = rule.Rule{
 		// to resolve variable.defs[0].node.init and checks isNonNullaryLiteral.
 		// We use TypeChecker symbol resolution which is more accurate (handles cross-file, type info).
 		checkIdentifier := func(expr *ast.Node, reportNode *ast.Node) {
+			// TypeChecker is nil for gap files (files in the program but not
+			// in typeInfoFiles). Without it we cannot resolve the identifier
+			// to its declaration — skip the check instead of panicking.
+			if ctx.TypeChecker == nil {
+				return
+			}
 			decl := utils.GetDeclaration(ctx.TypeChecker, expr)
 			if decl == nil {
 				return
@@ -167,7 +173,12 @@ var StylePropObjectRule = rule.Rule{
 						if nameNode == nil || nameNode.Kind != ast.KindIdentifier || nameNode.AsIdentifier().Text != "style" {
 							continue
 						}
-						// For shorthand { style }, resolve the value symbol to the variable declaration
+						// For shorthand { style }, resolve the value symbol to the variable declaration.
+						// TypeChecker is nil for gap files — skip the check in that case
+						// rather than panicking.
+						if ctx.TypeChecker == nil {
+							return
+						}
 						valueSymbol := ctx.TypeChecker.GetShorthandAssignmentValueSymbol(prop)
 						if valueSymbol != nil && len(valueSymbol.Declarations) > 0 {
 							decl := valueSymbol.Declarations[0]

